tests/http/test_client.py

```diff
@@ -0,0 +1,64 @@
+import pytest
+from httpx import ConnectTimeout, MockTransport, Request, Response, codes
```

Comment:
Maybe better to introduce something more powerful like
https://github.com/lundberg/respx?